client/rest-high-level/build.gradle

```diff
@@ -30,6 +30,7 @@ group = 'org.elasticsearch.client'
 archivesBaseName = 'elasticsearch-rest-high-level-client'
 
 restResources {
+  sourceSetName 'test'
```

Comment:
perhaps to make the migration less cumbersome we should default the rest resources plugin to continue to use the `test` source set, and only set it to `yamlRestTest` when the yaml rest test plugin is applied to a project.

Reply:
indeed.  I like this better. [`6289b00` (#56841)](https://github.com/elastic/elasticsearch/pull/56841/commits/6289b00b7fb71856968e2a6fa543202f4aa8cf7d)

I had to do a little bit of gymnastics to ensure that configuration that uses source set name is indeed lazily evaluated : [`6289b00` (#56841)](https://github.com/elastic/elasticsearch/pull/56841/commits/6289b00b7fb71856968e2a6fa543202f4aa8cf7d#diff-1e33902b5fc874b367bd346bd6e23790R68) , the dependency from within the `.create` of the task was causing the `.register` of the task to also eagerly evaluate which caused an incorrect source set name since we now set the source set name when the YamlRestTestPlugin is applied.

Reply:
I don't see why these gymnastics are necessary. We have the resources plugin define a default value and we override it in the `YamlRestTestPlugin`.

All the stuff in ` project.getPlugins().withType(YamlRestTestPlugin.class)` should just go in `YamlRestTestPlugin` itself.

Reply:
thanks @mark-vieira I removed the gymnastics as described. However, i did need to leave in the last afterEvalute else we have an ordering requirement between the application of the Java plugin and the RestResources plugin (due to the need to optionally create a dependency from the `processTestResources`)